src/main/kotlin/io/github/gradlenexus/publishplugin/internal/InvalidatingStagingRepositoryDescriptorRegistry.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin.internal
+
+import org.gradle.api.artifacts.repositories.ArtifactRepository
+import org.gradle.api.internal.artifacts.repositories.AbstractResolutionAwareArtifactRepository
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.util.concurrent.ConcurrentHashMap
+
+class InvalidatingStagingRepositoryDescriptorRegistry {
```

Comment:
I wonder, if `InvalidatingStagingRepositoryDescriptorRegistry` couldn't extend `StagingRepositoryDescriptorRegistry` (unless there are any technical issues with that)?

Assuming this is a temporary workaround, it would be easier to revert that in the future (currently, GitHub reports this class as a new one, and `StagingRepositoryDescriptorRegistry` as deleted, what can lead to harder access to the history of changes).